### PR TITLE
[MDS-5852] update the roles to administrative for deletion

### PR DESCRIPTION
**Pull request by isuru-aot**
## Objective 

[MDS-5852](https://bcmines.atlassian.net/browse/MDS-5852)

_Why are you making this change? Provide a short explanation and/or screenshots_

Update the roles required for `Delete` action to `core_edit_reports`.

Relevant info: https://discord.com/channels/976597427421323265/1252363484776235009

### CODE REVIEW

File: services/core-web/src/components/mine/Reports/MineReportTable.js [hunk 1]
```diff
@@ -62,7 +62,9 @@ const defaultProps = {
 export const MineReportTable = (props) => {
   const hideColumn = (condition) => (condition ? "column-hide" : "");
 
-  const userIsAdmin = useSelector((state) => userHasRole(state, USER_ROLES.role_admin));
+  const userIsAdministrative = useSelector((state) =>
+    userHasRole(state, USER_ROLES.role_mds_administrative_users)
```
Comment (taraepp) [inline services/core-web/src/components/mine/Reports/MineReportTable.js]: Just checking if the intention behind the ticket is to _switch_ the role required to access delete or if it should be to _add another_ role that can delete, did not seem obvious to me.

Comment (isuru-aot) [reply services/core-web/src/components/mine/Reports/MineReportTable.js:66]: @taraepp when a user has admin role, doesn't he has all the roles?

Comment (isuru-aot) [reply services/core-web/src/components/mine/Reports/MineReportTable.js]: @taraepp when a user has an admin role, doesn't he has all the roles?

Comment (isuru-aot) [reply services/core-web/src/components/mine/Reports/MineReportTable.js]: Had a discussion around this here: https://discord.com/channels/976597427421323265/1252363484776235009

**Review by simensma-fresh — APPROVED** [review]

**Review by henryoforeh-dev — APPROVED** [review]
Nice work!

**Pull request merged**